fedimint-client/src/lib.rs

```diff
@@ -782,6 +783,64 @@ impl Client {
                 .collect(),
         }
     }
+
+    /// Load the common api versions to use from cache and start a background
+    /// process to refresh them.
+    ///
+    /// This is a compromise so we not have to wait for version discovery to
+    /// complete every time a [`Client`] is being built.
+    async fn load_and_refresh_common_api_version_static(
+        config: &ClientConfig,
+        module_gens: &ModuleGenRegistry<DynClientModuleGen>,
+        api: &DynGlobalApi,
+        db: &Database,
+    ) -> anyhow::Result<ApiVersionSet> {
+        if let Some(v) = db
+            .begin_transaction()
+            .await
+            .get_value(&CachedApiVersionSetKey)
+            .await
+        {
+            let config = config.clone();
+            let module_gens = module_gens.clone();
+            let api = api.clone();
+            let db = db.clone();
+            tokio::task::spawn(async move {
+                if let Err(e) =
+                    Self::refresh_common_api_version_static(&config, &module_gens, &api, &db).await
+                {
+                    warn!("Failed to discover common api versions: {e}");
+                }
+            });
```

Comment:
This isn't in a task group because it's short-lived/potentially spawned many times? Also, didn't we have some WASM-compat `spawn` method?

Reply:
Yeah, I don't think we need to wait for it. It will either succeed or not. It's kind of a best effort side quest.

And yeah, I think I should use fedimint_core::task instead.